src/task/blocking.rs

```diff
@@ -47,18 +58,34 @@ lazy_static! {
     };
 }
 
-// Create up to MAX_THREADS dynamic blocking task worker threads.
-// Dynamic threads will terminate themselves if they don't
-// receive any work after between one and ten seconds.
-fn maybe_create_another_blocking_thread() {
-    // We use a `Relaxed` atomic operation because
-    // it's just a heuristic, and would not lose correctness
-    // even if it's random.
-    let workers = DYNAMIC_THREAD_COUNT.load(Ordering::Relaxed);
-    if workers >= MAX_THREADS {
-        return;
+fn calculate_dispatch_frequency() {
+    // Calculate current message processing rate here
+    let current_freq = FREQUENCY.fetch_sub(1, Ordering::Relaxed);
+    let avr_freq = AVR_FREQUENCY.load(Ordering::Relaxed);
+    let current_pool_size = CURRENT_POOL_SIZE.load(Ordering::Relaxed);
```

Comment:
For defense in depth and Relaxed raciness protection:
`std::cmp::max(LOW_WATER_MARK, CURRENT_POOL_SIZE.load(Ordering::Relaxed)`